sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateSampleDao.java

```diff
@@ -339,15 +346,39 @@ public boolean aliasExists(String alias) throws IOException {
 
   @Override
   public Collection<Sample> listBySearch(String querystr) throws IOException {
-    Query query = currentSession().createQuery(
-        "from SampleImpl where identificationBarcode like :query or name LIKE :query or alias like :query or description like :query or scientificName like :query");
-    query.setString("query", querystr);
+    Criteria criteria = currentSession().createCriteria(Sample.class);
+    criteria.add(Restrictions.or(Restrictions.ilike("identificationBarcode", querystr + "%"), Restrictions.ilike("name", querystr + "%"),
+        Restrictions.ilike("alias", querystr + "%")));
```

Comment:
Consider adding a wildcard to the front of the string as well. Right now searching for `KGMT` will match `KGMT_0015`, but searching for `0015` will not. Since the query string is not escaped, it's currently possible to use SQL wildcards in queries (this is not something a general user will do.) So searching for `%0015` will match against `KGMT_0015`.